### PR TITLE
document decision about cryptographic key handling (crypto plugin)

### PR DESCRIPTION
**Pull request by petermax2**
adding a decision as requested on the mailing list.

### CODE REVIEW

File: doc/decisions/cryptograhic_key_handling.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+operations, thus:
+
+- the combination of key and IV must not be used twice for symmetric encryption
+- the key and IV material must be as random as possible (using an SNRG)
+- the key is not exposed to other Elektra modules or the user
+
+## Assumptions
+
+- The user knows or is willing to learn how to use GPG.
+- The user is willing to configure `gpg` and `pinentry` on her system, as it does not always run out of the box (depending on the distribution)
```
Comment (markus2330) [inline doc/decisions/cryptograhic_key_handling.md]: Can you more specific here? Especially the gpg part: Which setup is required? Do you mean creating a key-pair?

File: doc/decisions/cryptograhic_key_handling.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+## Considered Alternatives
+
+We considered passing the key and IV in form of plugin configuration and meta-keys, but this approach violates our constraints.
```
Comment (markus2330) [inline doc/decisions/cryptograhic_key_handling.md]: Give some more alternatives, e.g. having password as plugin config, directly using gpg-agent or pcscd,...

Also argument better: which specific constraint is violated.

File: doc/decisions/cryptograhic_key_handling.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+## Decision
+
+### General Approach
+
+The introduction of a KDF will enable the user to provide a single password, which is used to derive the key and the IV for the cryptographic operations. Both OpenSSL and libgcrypt have built-in support for the PBKDF2 (see RFC 2898).
+
+Instead of having the keys `crypto/iv` and `crypto/key` within the plugin configuration, we will have `crypto/password` for the moment. In the future there will be other ways of requesting this password (like the pgp-agent but this will be another issue).
```
Comment (markus2330) [inline doc/decisions/cryptograhic_key_handling.md]: I think this is old text

**Comment by petermax2** [comment]
I updated the text. You were right about the old text. I am going to update the description in the ticket too.

**Comment by markus2330** [comment]
Can it be merged now?

**Comment by petermax2** [comment]
Yes, it is ready if there are not any further points of critique.

**Pull request merged**